### PR TITLE
Laboratory work #2, Sofya Kostina- 21FPL2

### PR DESCRIPTION
**Pull request by d07l**

### CODE REVIEW

File: lab_2_keywords_cooccurrence/main.py [hunk 1]
```diff
@@ -29,7 +41,34 @@ def extract_candidate_keyword_phrases(phrases: Sequence[str], stop_words: Sequen
...
+            if word not in stop_words:
+                sublist.append(word)
+            else:
+                if sublist:
+                    sublist = tuple(sublist)
+                    keywords.append(sublist)
+                    sublist = []
+                else:
+                    continue
+        sublist = tuple(sublist)
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:67]: you need to do it only if `sublist` is not empty

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_keywords_cooccurrence/main.py [hunk 2]
```diff
@@ -17,7 +17,19 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(text, str) or not text:
+        return None
+    punctuation = """.,;:¡!¿?…⋯‹›«»/\\"“”[]()⟨⟩}{&|-–~—"""
+    for i in text:
+        if i in punctuation:
+            text = text.replace(i, 'splt')
```
[the remark this answers is not in the document]
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py]: fixed

File: lab_2_keywords_cooccurrence/main.py [hunk 3]
```diff
@@ -17,7 +17,19 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
...
+    splited_text = text.split('splt')
+    tokens = []
+    for token in splited_text:
+        token = token.strip()
+        if token != '':
```
[the remark this answers is not in the document]
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py]: fixed

File: lab_2_keywords_cooccurrence/main.py (hunk 1 above)
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py]: fixed (I guess)

File: lab_2_keywords_cooccurrence/main.py [hunk 4]
```diff
@@ -17,7 +17,19 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(text, str) or not text:
+        return None
+    punctuation = """.,;:¡!¿?…⋯‹›«»/\\"“”[]()⟨⟩}{&|-–~—"""
+    for i in text:
+        if i in punctuation:
+            text = text.replace(i, '.')
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:25]: remove `.` from `punctuation` if you use `.` as a special character

File: lab_2_keywords_cooccurrence/main.py [hunk 5]
```diff
@@ -29,7 +41,33 @@ def extract_candidate_keyword_phrases(phrases: Sequence[str], stop_words: Sequen
...
+        return None
+
+    phrases = [i.lower() for i in phrases]
+    tokens = []
+    keywords = []
+    for sentence in phrases:
+        sublist = []
+        element = ""
+        for symbol in sentence:
+            element += symbol
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:54]: maybe you can use `str.join`

File: lab_2_keywords_cooccurrence/main.py [hunk 6]
```diff
@@ -29,7 +41,33 @@ def extract_candidate_keyword_phrases(phrases: Sequence[str], stop_words: Sequen
...
+        words = element.split()
+        tokens.append(words)
+        for word in words:
+            if word not in stop_words:
+                sublist.append(word)
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:59]: add `continue` and remove `else`

File: lab_2_keywords_cooccurrence/main.py [hunk 7]
```diff
@@ -55,7 +101,19 @@ def calculate_word_degrees(candidate_keyword_phrases: KeyPhrases,
...
+            or not candidate_keyword_phrases or not content_words:
+        return None
+    wd_dict = {}
+    for word in content_words:
+        wdegree = 0
+        for phrase in candidate_keyword_phrases:
+            if word in phrase:
+                wdegree += len(phrase)
+            else:
+                continue
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:114]: it is not needed

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_keywords_cooccurrence/main.py (hunk 4 above)
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py:25]: removed

File: lab_2_keywords_cooccurrence/main.py (hunk 5 above)
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py]: done

File: lab_2_keywords_cooccurrence/main.py (hunk 6 above)
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py:57]: fixed

File: lab_2_keywords_cooccurrence/main.py (hunk 7 above)
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py:111]: removed

File: lab_2_keywords_cooccurrence/main.py [hunk 8]
```diff
@@ -123,7 +211,56 @@ def extract_candidate_keyword_phrases_with_adjoining(candidate_keyword_phrases:
...
+
+    phrases_tokenized = []
+    for i in phrases:
+        i = i.split(' ')
+        phrases_tokenized.append(i)
+    phrases_tokenized = [item.lower() for i in phrases_tokenized for item in i if item]
+
+    new_keywords = []
+    for k, v in pairs_dict.items():
+        if v >= 2:
```
[the remark this answers is not in the document]
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py]: fixed

File: lab_2_keywords_cooccurrence/main.py [hunk 9]
```diff
@@ -123,7 +211,56 @@ def extract_candidate_keyword_phrases_with_adjoining(candidate_keyword_phrases:
...
+            tokenized_key = ' '.join(k)
+            tokenized_key = tokenized_key.split()
+
+            for ind, word in enumerate(phrases_tokenized):
+                list_with_stopword = []
+                if word in tokenized_key:
+                    list_with_stopword.append(phrases_tokenized[ind:ind + (len(tokenized_key) + 1)])
+
+                if list_with_stopword:
```
[the remark this answers is not in the document]
Comment (d07l) [reply lab_2_keywords_cooccurrence/main.py]: fixed

File: lab_2_keywords_cooccurrence/main.py (hunk 7 above)
Comment (artyomtugaryov) [reply lab_2_keywords_cooccurrence/main.py:112]: but not pushed?

File: lab_2_keywords_cooccurrence/main.py [hunk 10]
```diff
@@ -40,7 +75,16 @@ def calculate_frequencies_for_content_words(candidate_keyword_phrases: KeyPhrase
...
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(candidate_keyword_phrases, list) or not candidate_keyword_phrases:
+        return None
+    keywords_list = []
+    for keyphrase in candidate_keyword_phrases:
+        for word in keyphrase:
+            keywords_list.append(word)
+    frequencies = {}
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:84]: one line

File: lab_2_keywords_cooccurrence/main.py [hunk 11]
```diff
@@ -123,7 +208,53 @@ def extract_candidate_keyword_phrases_with_adjoining(candidate_keyword_phrases:
...
+    new_keywords = []
+    for k, value in pairs_dict.items():
+        if value < 2:
+            continue
+        token_key = ' '.join(k)
+        tokenized_key = token_key.split()
+
+        for ind, word in enumerate(phrases_tokenized):
+            list_with_stopword = []
+            if word in tokenized_key:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:233]: if word not in ...:
 continue

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — APPROVED** [review]

**Pull request closed without merging**